launcher/minecraft/mod/tasks/ModFolderLoadTask.cpp

```diff
@@ -0,0 +1,53 @@
+#include "ModFolderLoadTask.h"
+#include <QDebug>
+
+#include "Application.h"
+#include "minecraft/mod/MetadataHandler.h"
+
+ModFolderLoadTask::ModFolderLoadTask(QDir& mods_dir, QDir& index_dir) 
+    : m_mods_dir(mods_dir), m_index_dir(index_dir), m_result(new Result())
+{}
+
+void ModFolderLoadTask::run()
+{
+    if (!APPLICATION->settings()->get("DontUseModMetadata").toBool()) {
+        // Read metadata first
+        getFromMetadata();
+    }
+
+    // Read JAR files that don't have metadata
+    m_mods_dir.refresh();
+    for (auto entry : m_mods_dir.entryInfoList()) {
+        Mod mod(entry);
+        if(m_result->mods.contains(mod.internal_id())){
+            m_result->mods[mod.internal_id()].setStatus(ModStatus::Installed);
+        }
+        else {
+            m_result->mods[mod.internal_id()] = mod;
+            m_result->mods[mod.internal_id()].setStatus(ModStatus::NoMetadata);
+        }
+    }
+
+    emit succeeded();
+}
+
+void ModFolderLoadTask::getFromMetadata()
+{
+    m_index_dir.refresh();
+    for (auto entry : m_index_dir.entryList()) {
+        // QDir::Filter::NoDotAndDotDot seems to exclude all files for some reason...
+        if (entry == "." || entry == "..")
+            continue;
+
```

Comment:
`QDir::Files` should work, as `.` and `..` aren't files. If not, just do `QDir::Files | QDir::NoDotAndDotDot`
```suggestion
    for (auto entry : m_index_dir.entryList(QDir::Files)) {
```